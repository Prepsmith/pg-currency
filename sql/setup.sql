--
-- first, define the datatype.  Turn off echoing so that expected file
-- does not depend on contents of citext.sql.
--
SET client_min_messages = warning;
\set ECHO none
\i currency.sql
RESET client_min_messages;
\set ECHO all

create table wp_currencies (
       code char(3),
       num int,
       minor int,
       description text
);

\copy wp_currencies from 'data/iso4217.data'

INSERT INTO currency_rate
       (currency_number, currency_code, name, symbol, minor, rate)
VALUES
	(0, 'BTC', 'Bitcoin', 'à¸¿', 2, 1);

INSERT INTO currency_rate
       (currency_number, currency_code, name, minor, rate)
SELECT
	num, code, description, minor, (random()+0.5)^3
from
	wp_currencies;

-- some rates for testing
update currency_rate set rate = 3 where currency_code = 'NZD';
update currency_rate set rate = 4 where currency_code = 'USD';
update currency_rate set rate = 6 where currency_code = 'EUR';
update currency_rate set rate = 7 where currency_code = 'GBP';
