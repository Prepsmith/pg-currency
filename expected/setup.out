--
-- first, define the datatype.  Turn off echoing so that expected file
-- does not depend on contents of citext.sql.
--
SET client_min_messages = warning;
SET
\set ECHO none
SET
CREATE TYPE
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE TYPE
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE FUNCTION
CREATE OPERATOR
CREATE OPERATOR
CREATE OPERATOR
CREATE OPERATOR
CREATE OPERATOR
CREATE OPERATOR
CREATE CAST
CREATE CAST
CREATE OPERATOR CLASS
CREATE OPERATOR CLASS
COMMENT
CREATE TABLE
RESET
create table wp_currencies (
       code char(3),
       num int,
       minor int,
       description text
);
CREATE TABLE
\copy wp_currencies from 'data/iso4217.data'
INSERT INTO currency_rate
       (currency_number, currency_code, name, symbol, minor, rate)
VALUES
	(0, 'BTC', 'Bitcoin', 'à¸¿', 2, 1);
INSERT 0 1
INSERT INTO currency_rate
       (currency_number, currency_code, name, minor, rate)
SELECT
	num, code, description, minor, (random()+0.5)^3
from
	wp_currencies;
INSERT 0 170
-- some rates for testing
update currency_rate set rate = 3 where currency_code = 'NZD';
UPDATE 1
update currency_rate set rate = 4 where currency_code = 'USD';
UPDATE 1
update currency_rate set rate = 6 where currency_code = 'EUR';
UPDATE 1
update currency_rate set rate = 7 where currency_code = 'GBP';
UPDATE 1
